### PR TITLE
Introduce PSVersionAttribute

### PR DESCRIPTION
**Pull request by iSazonov**
# PR Summary

Address #11667

Added new PSVersionAttribute class.
Added Version property to FunctionInfo and ExternalScriptInfo classes.

## PR Context

```powershell
function TestFunction
{
   [cmdletbinding()]
   [OutputType([int])]
   [PSVersionAttribute(1,2)]
   Param ($Parameter1)

   "test"
}

(Get-Command TestFunction).Version
```
Output:
```powershell
Major  Minor  Build  Revision
-----  -----  -----  --------
1      2      -1     -1
```
## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [x] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

File: src/System.Management.Automation/engine/Attributes.cs [hunk 1]
```diff
@@ -515,6 +515,46 @@ public string[] ParameterSetName
         private string[] _parameterSetName;
```
Comment (PoshChan) [inline src/System.Management.Automation/engine/Attributes.cs]: @iSazonov, your last commit had 1 failures in `PowerShell-CI-linux`
Type accelerators.BuiltIn Accelerators.Should have all the type accelerators
```powershell
Expected 99, but got 100.
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Language/Parser/TypeAccelerator.Tests.ps1: line 441
441:             $TypeAccelerators.Count | Should -Be $totalAccelerators
```

Comment (PoshChan) [inline src/System.Management.Automation/engine/Attributes.cs]: @iSazonov, your last commit had 1 failures in `PowerShell-CI-windows`
Type accelerators.BuiltIn Accelerators.Should have all the type accelerators
```powershell
Expected 104, but got 105.
at <ScriptBlock>, D:\a\1\s\test\powershell\Language\Parser\TypeAccelerator.Tests.ps1: line 441
441:             $TypeAccelerators.Count | Should -Be $totalAccelerators
```

Comment (PoshChan) [inline src/System.Management.Automation/engine/Attributes.cs]: @iSazonov, your last commit had 1 failures in `PowerShell-CI-macos`
Type accelerators.BuiltIn Accelerators.Should have all the type accelerators
```powershell
Expected 99, but got 100.
at <ScriptBlock>, /Users/runner/runners/2.164.6/work/1/s/test/powershell/Language/Parser/TypeAccelerator.Tests.ps1: line 441
441:             $TypeAccelerators.Count | Should -Be $totalAccelerators
```

**Comment by doctordns** [comment]
Should this attribute be extended to cover script files too?

**Comment by ThomasNieto** [comment]
If so there may need to be some discussion about PSGet 3.0 and how versions are going be defined with the old method and this new way.

File: src/System.Management.Automation/engine/Attributes.cs (hunk 1 above)
Comment (PoshChan) [inline src/System.Management.Automation/engine/Attributes.cs]: @iSazonov, your last commit had 1 failures in `PowerShell-CI-linux`
Type accelerators.BuiltIn Accelerators.Should have a type accelerator for: PSVersion
```powershell
Expected [PSVersionAttribute], but got $null.
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Language/Parser/TypeAccelerator.Tests.ps1: line 450
450:             $TypeAcceleratorsType::Get[$Accelerator] | Should -Be ($Type)
```

Comment (PoshChan) [inline src/System.Management.Automation/engine/Attributes.cs]: @iSazonov, your last commit had 1 failures in `PowerShell-CI-macos`
Type accelerators.BuiltIn Accelerators.Should have a type accelerator for: PSVersion
```powershell
Expected [PSVersionAttribute], but got $null.
at <ScriptBlock>, /Users/runner/runners/2.164.6/work/1/s/test/powershell/Language/Parser/TypeAccelerator.Tests.ps1: line 450
450:             $TypeAcceleratorsType::Get[$Accelerator] | Should -Be ($Type)
```

Comment (PoshChan) [inline src/System.Management.Automation/engine/Attributes.cs]: @iSazonov, your last commit had 1 failures in `PowerShell-CI-windows`
Type accelerators.BuiltIn Accelerators.Should have a type accelerator for: PSVersion
```powershell
Expected [PSVersionAttribute], but got $null.
at <ScriptBlock>, D:\a\1\s\test\powershell\Language\Parser\TypeAccelerator.Tests.ps1: line 450
450:             $TypeAcceleratorsType::Get[$Accelerator] | Should -Be ($Type)
```

**Comment by iSazonov** [comment]
@doctordns I added support for external script.

File: src/System.Management.Automation/engine/Attributes.cs [hunk 2]
```diff
@@ -515,6 +515,55 @@ public string[] ParameterSetName
...
```
Comment (PoshChan) [inline src/System.Management.Automation/engine/Attributes.cs]: @iSazonov, your last commit had 1 failures in `PowerShell-CI-linux`
Invoke-WebRequest tests.Denial of service.Charset Parsing
```powershell
Expected the actual value to be greater than 5, but got 0.214421204196174.
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1: line 1961
1961:             $pathologicalRatio | Should -BeGreaterThan 5
```

**Comment by iSazonov** [comment]
I hope PowerShell Committee reviews the PR and related issue and makes a conclusion about the new attribute.

**Comment by JamesWTruher** [comment]
what happens if this is used in a function which is part of module?

**Comment by iSazonov** [comment]
If a function is in a module, the module version should be used.

**Comment by SteveL-MSFT** [comment]
@iSazonov please see the comments by @PowerShell/powershell-committee in the linked issue

**Comment by iSazonov** [comment]
Close based on PowerShell Committee conclusion but keep the branch for future.

**Pull request closed without merging**